src/Catty/Functions/Text/LetterFunction.swift

```diff
@@ -0,0 +1,51 @@
+/**
+ *  Copyright (C) 2010-2018 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class LetterFunction: DoubleParameterStringFunction {
+    static var tag = "LETTER"
+    static var name = "letter"
+    static var defaultValue = ""
+    static var requiredResource = ResourceType.noResources
+    static var isIdempotent = true
+    static let position = 220
+    
+    static func firstParameter() -> FunctionParameter {
+        return .number(defaultValue: 1)
+    }
+    
+    static func secondParameter() -> FunctionParameter {
+        return .string(defaultValue: "hello world")
+    }
+    
+    func value(firstParameter: AnyObject?, secondParameter: AnyObject?) -> String {
+        guard let number = firstParameter as? Int,
+            let text = secondParameter as? String else {
+                return type(of: self).defaultValue
+        }
+        let index = text.index(text.startIndex, offsetBy: number - 1)
```

Comment:
According to Apple's documentation of the index function:

`/// The value passed as `n` must not offset `i` beyond the bounds of the collection.`

Please add a test where the first parameter is negative (e.g. -1). You will see that the app is crashing.